tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -403,6 +404,156 @@ def _query_rules(self, resource):
+                ip-rules:
+                  - 19.0.0.0/16
+                  - 20.0.1.2
+
+
+     Cosmos firewalls are disabled by simply configuring them with empty values.
+     We can do this with an empty action, which defaults to ``append: False``.
+
+     .. code-block:: yaml
+
+        policies:
+          - name: cosmos-firewall-clear
+            resource: azure.cosmosdb
+            filters:
+              # The firewall is enabled
+              - not:
+                - type: value
+                  key: properties.ipRangeFilter
+                  value: empty
+            actions:
+              - type: set-firewall-rules
+
+
+     """
+
+    schema = type_schema(
+        'set-firewall-rules',
+        rinherit=SetFirewallAction.schema,
+        **{
+            'bypass-rules': {'type': 'array', 'items': {
+                'enum': ['Portal', 'AzureCloud']}},
+        }
+    )
+
+    def __init__(self, data, manager=None):
+        super(CosmosSetFirewallAction, self).__init__(data, manager)
+        self._log = logging.getLogger('custodian.azure.cosmosdb')
+        self.rule_limit = 1000
+
+    def _process_resource(self, resource):
+        existing_ip = filter(None, resource['properties'].get('ipRangeFilter', '').split(','))
+        rules = self._build_ip_rules(existing_ip, self.data.get('ip-rules', []))
+
+        # Cosmos DB does not have real bypass
+        # instead the portal UI adds these values to your
+        # rules filter when you check the box.
+        bypass = self.data.get('bypass-rules', [])
+        if 'Portal' in bypass:
+            rules.extend(['104.42.195.92',
+                          '40.76.54.131',
+                          '52.176.6.30',
+                          '52.169.50.45',
+                          '52.187.184.26'])
+        if 'AzureCloud' in bypass:
+            rules.append('0.0.0.0')
+
+        # If the user has too many rules log and skip
+        if len(rules) > self.rule_limit:
+            self._log.error("Skipped updating firewall for %s. "
+                            "%s exceeds maximum rule count of %s." %
+                            (resource['name'], len(rules), self.rule_limit))
+            return
+
+        # Add VNET rules
+        existing_vnet = \
+            [r['id'] for r in resource['properties'].get('virtualNetworkRules', [])]
+        vnet_rules = self._build_vnet_rules(existing_vnet,
+                                            self.data.get('virtual-network-rules', []))
+
+        # Workaround for bug https://git.io/fjFLY
+        resource['properties']['locations'] = []
+        for loc in resource['properties'].get('readLocations'):
+            resource['properties']['locations'].append(
+                {'location_name': loc['locationName'],
+                 'failover_priority': loc['failoverPriority'],
+                 'is_zone_redundant': loc.get('isZoneRedundant', False)})
+
+        resource['properties']['ipRangeFilter'] = ','.join(rules)
```

Comment:
Can we rename to ip_rules so down here we know how these are different than vnet_rules?